BossMod/Components/CastHint.cs

```diff
@@ -27,6 +28,40 @@ public override void OnCastStarted(BossModule module, Actor caster, ActorCastInf
                 _casters.Add(caster);
         }
 
+        public override void OnCastFinished(BossModule module, Actor caster, ActorCastInfo spell)
+        {
+            if (spell.Action == WatchedAction)
+                _casters.Remove(caster);
+        }
+    }
+    public class EnrageCastHint : CastCounter
```

Comment:
I thought it would be useful when the exact timeline of the fight is unknown. For example if enrage cast is %hp based instead of time based because you simply have no replays that go until enrage start